### ISSUE TITLE
Multi-level menu entries (nested `SnippetViewSetGroup`)

### ISSUE THREAD

**Issue by tbrlpld** [wagtail/wagtail#10786]
### Is your proposal related to a problem?

Often you have multiple models that related to a similar topic, this happens in particular when providing filtering capabilities for content along multiple taxonomies. E.g. event type, event location, ...

Currently, we have the option to create a top level menu item for model admin and then list all related items below that (e.g. "taxonomies"). The list can become very long and contain several options that need to do name spacing in the label. E.g. 
* Taxonomies 
  * Event type
  * Event location
  * Person type
  * Person role
  * ..

The long list with the name spacing becomes hard to parse. 

### Describe the solution you'd like

I think it would be beneficial to be able to create multiple levels in the sidebar menu, to allow grouping of related entries on more than one live. 

* Taxonomies
  * Events
    * Type
    * Location
    * ...
  * People
    * Type
    * Role
    * ...

This could see this being possible by allowing `SnippetViewSetGroup` to contain other `SnippetViewSetGroup` classes in the `items` iterable. 

### Describe alternatives you've considered

The flat list with name spaces labels is the currently used approach.

**Comment by laymonage** [comment]
Thank you for the suggestion!

ModelAdmin has been deprecated as of Wagtail 5.1, but the same enhancement request can be applied to `SnippetViewSetGroup`, so I've updated the issue accordingly.

For the record, we're also working to bring snippets' features such as menu item registration to Wagtail's `ModelViewSet` (see #10740 for more details) so that we have a replacement for ModelAdmin for models that don't fit into the "snippets" terminology.

As part of this work, I reworked the menu item registration mechanism in #10767 and with the changes in that PR it is possible to do so:

<img width="1003" alt="image" src="https://github.com/wagtail/wagtail/assets/6379424/6baf04e3-aa8a-493d-b9e9-a29f72b2ac6d">

The styling would probably need some work if we want to support this officially, though.

**Comment by laymonage** [comment]
As noted above, this technically already works (somewhat). However, we have a few more improvements that we can make to have this officially supported:

- Make sure the "active path" highlight also works for items inside the submenus (not just the top-level one).
  - This currently doesn't work. It's likely just missing the styles, because from what I can tell, you can already derive the active submenu item from the component state (see `activePath` and `navigationPath`).
- Make sure that the close button on mobile allows you to go "one level up" when you're inside a deeply nested submenu, instead of taking you to the main menu.
  - This was discussed in https://github.com/wagtail/wagtail/pull/12086#pullrequestreview-2409884722, but I couldn't get it to work for some reason.

If anyone wants to pick it up, I highly suggest reading the review comments in the above PR to make sure that the solution also works well for mobile screens.

And then, we'll need to update the documentation on `ViewSetGroup.items` to note that you can also put `ViewSetGroup`s in there. One thing to note is that I had to add `add_to_admin_menu = False` for the inner menus, otherwise they also show up at the top-level. It would be nice if `WagtailMenuRegisterableGroup.add_to_admin_menu` can automatically detect this and return `False` for menu groups that are not at the top-level.

I guess we'll also need to double check that nesting `ViewSetGroup`, `ModelViewSetGroup`, and `SnippetViewSetGroup` works. This may also involve figuring out a better approach for when you want a group to contain a mix of regular `ViewSet`, `ModelViewSet`, and `SnippetViewSet`. See also https://github.com/wagtail/wagtail/discussions/10446#discussioncomment-8780760

**Comment by laymonage** [comment]
Unfortunately this broke in 6.4 due to https://github.com/wagtail/wagtail/pull/12526#discussion_r1941248784. I think we should fix the basic functionality in a 6.4.x patch release, but the above improvements should be for a future feature release.

**Comment by robmoorman** [comment]
Found this to be relevant, keen to pick this up for a solution. We've also discussed the option to add a Divider.

Slack: https://wagtailcms.slack.com/archives/C81FGJR2S/p1775636152391079

**Comment by thibaudcolas** [comment]
that’d be great @robmoorman! Our next feature freeze is in 5 days if you want to aim for that 😬
